Next openocd_install target incompatible stlinkv2 config files

## Summary

OpenOCD appears to have merged stlink-type debug adaptors into the "hla" target. The in-tree "make openocd_install" target downloads and builds v0.6.1 of openocd, which appears not to have the 'hla' debug adaptor supported.

This results in the make ef_XXX_program targets failing to program the target board. 

Tested version: taulabs next,
commit-id:d4754e0c99a53ee49dc7b5c8cd7941d164991774
## Steps to reproduce

```
make openocd_install
make ef_flyingf4_target
```
## Expected result

openocd programs board
## Observed result

```
user@user-VirtualBox:/tmp/TauLabs$ make ef_flyingf4_program
make[1]: Nothing to be done for `first'.
make[1]: Nothing to be done for `bin'.
- Tau Labs UAVObject Generator -
Done: processed 101 XML files and generated 101 objects with no ID collisions. Total size of the data fields is 3268 bytes.
generating flight code
 FWINFO      build/fw_flyingf4/fw_flyingf4.bin.firmwareinfo.c
 PADDING     build/ef_flyingf4/ef_flyingf4.fw_pre.pad
 PADDING     build/ef_flyingf4/ef_flyingf4.fw_post.pad
 FLASH_IMG   build/ef_flyingf4/ef_flyingf4.bin
 JTAG-PGM    build/ef_flyingf4/ef_flyingf4.bin
Open On-Chip Debugger 0.6.1 (2013-09-04-11:44)
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.sourceforge.net/doc/doxygen/bugs.html
debug_level: 0
Error: The specified debug interface was not found (hla)
The following debug interfaces are available:
1: ft2232
2: stlink
Runtime Error: /tmp/TauLabs/flight/Project/OpenOCD/stlink-v2.cfg:5:
in procedure 'script'
at file "embedded:startup.tcl", line 58
in procedure 'interface' called at file "/tmp/TauLabs/flight/Project/OpenOCD/stlink-v2.cfg", line 5
make[1]: *** [program] Error 1
make: *** [ef_flyingf4_program] Error 2
```
## Possible fixes?

Update openocd to current v0.8.0 which supports hla target referenced in configfile in-tree. 
-or-
Modify in-tree config file to use legacy stlink target.

Note that this target works perfectly fine with v0.8.0 installed on system, if available.

--- a/make/tools.mk
+++ b/make/tools.mk
@@ -102,50 +102,6 @@ arm_sdk_install: arm_sdk_clean
 arm_sdk_clean:
 	$(V1) [ ! -d "$(ARM_SDK_DIR)" ] || $(RM) -r $(ARM_SDK_DIR)
 
-# Set up openocd tools
-OPENOCD_DIR       := $(TOOLS_DIR)/openocd
-OPENOCD_WIN_DIR   := $(TOOLS_DIR)/openocd_win
-OPENOCD_BUILD_DIR := $(DL_DIR)/openocd-build
-
-.PHONY: openocd_install
-openocd_install: | $(DL_DIR) $(TOOLS_DIR)
-openocd_install: OPENOCD_URL  := http://sourceforge.net/projects/openocd/files/openocd/0.6.1/openocd-0.6.1.tar.bz2/download
-openocd_install: OPENOCD_FILE := openocd-0.6.1.tar.bz2
-openocd_install: OPENOCD_OPTIONS := --prefix="$(OPENOCD_DIR)" --enable-stlink
-
-ifeq ($(OPENOCD_FTDI), yes)
-openocd_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --enable-ft2232_libftdi
-endif
-
-openocd_install: openocd_clean
-        # download the source only if it's newer than what we already have
-	$(V1) wget -N -P "$(DL_DIR)" --trust-server-name "$(OPENOCD_URL)"
-
-        # extract the source
-	$(V1) [ ! -d "$(OPENOCD_BUILD_DIR)" ] || $(RM) -r "$(OPENOCD_BUILD_DIR)"
-	$(V1) mkdir -p "$(OPENOCD_BUILD_DIR)"
-	$(V1) tar -C $(OPENOCD_BUILD_DIR) -xjf "$(DL_DIR)/$(OPENOCD_FILE)"
-
-        # apply patches
-	$(V0) @echo " PATCH        $(OPENOCD_DIR)"
-	$(V1) ( \
-	  cd $(OPENOCD_BUILD_DIR)/openocd-0.6.1 ; \
-	  patch -p1 < $(ROOT_DIR)/flight/Project/OpenOCD/0001-armv7m-remove-dummy-FP-regs-for-new-gdb.patch ; \
-	  patch -p1 < $(ROOT_DIR)/flight/Project/OpenOCD/0002-rtos-add-stm32_stlink-to-FreeRTOS-targets.patch ; \
-	)
-
-        # build and install
-	$(V1) mkdir -p "$(OPENOCD_DIR)"
-	$(V1) ( \
-	  cd $(OPENOCD_BUILD_DIR)/openocd-0.6.1 ; \
-	  ./configure $(OPENOCD_OPTIONS) ; \
-	  $(MAKE) --silent ; \
-	  $(MAKE) --silent install ; \
-	)
-
-        # delete the extracted source when we're done
-	$(V1) [ ! -d "$(OPENOCD_BUILD_DIR)" ] || $(RM) -rf "$(OPENOCD_BUILD_DIR)"
-
 .PHONY: ftd2xx_install
 
 FTD2XX_DIR := $(DL_DIR)/ftd2xx
@@ -194,19 +150,18 @@ libusb_win_clean:
 	$(V0) @echo " CLEAN        $(LIBUSB_WIN_DIR)"
 	$(V1) [ ! -d "$(LIBUSB_WIN_DIR)" ] || $(RM) -r "$(LIBUSB_WIN_DIR)"
 
-.PHONY: openocd_git_win_install
+.PHONY: openocd_win_install
 
-openocd_git_win_install: | $(DL_DIR) $(TOOLS_DIR)
-openocd_git_win_install: OPENOCD_URL  := git://git.code.sf.net/p/openocd/code
-openocd_git_win_install: OPENOCD_REV  := cf1418e9a85013bbf8dbcc2d2e9985695993d9f4
-openocd_git_win_install: OPENOCD_OPTIONS := 
+openocd_win_install: | $(DL_DIR) $(TOOLS_DIR)
+openocd_win_install: OPENOCD_URL  := git://git.code.sf.net/p/openocd/code
+openocd_win_install: OPENOCD_REV  := cf1418e9a85013bbf8dbcc2d2e9985695993d9f4
+openocd_win_install: OPENOCD_OPTIONS := 
 
 ifeq ($(OPENOCD_FTDI), yes)
-openocd_git_win_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --enable-ft2232_ftd2xx --with-ftd2xx-win32-zipdir=$(FTD2XX_DIR)
+openocd_win_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --enable-ft2232_ftd2xx --with-ftd2xx-win32-zipdir=$(FTD2XX_DIR)
 endif
 
-openocd_git_win_install: openocd_win_clean libusb_win_install ftd2xx_install
-
+openocd_win_install: openocd_win_clean libusb_win_install ftd2xx_install
         # download the source
 	$(V0) @echo " DOWNLOAD     $(OPENOCD_URL) @ $(OPENOCD_REV)"
 	$(V1) [ ! -d "$(OPENOCD_BUILD_DIR)" ] || $(RM) -rf "$(OPENOCD_BUILD_DIR)"
@@ -250,24 +205,27 @@ openocd_win_clean:
 	$(V0) @echo " CLEAN        $(OPENOCD_WIN_DIR)"
 	$(V1) [ ! -d "$(OPENOCD_WIN_DIR)" ] || $(RM) -r "$(OPENOCD_WIN_DIR)"
 
-.PHONY: openocd_git_install
+# Set up openocd tools
+OPENOCD_DIR       := $(TOOLS_DIR)/openocd
+OPENOCD_WIN_DIR   := $(TOOLS_DIR)/openocd_win
+OPENOCD_BUILD_DIR := $(DL_DIR)/openocd-build
 
-openocd_git_install: | $(DL_DIR) $(TOOLS_DIR)
-openocd_git_install: OPENOCD_URL     := git://git.code.sf.net/p/openocd/code
-openocd_git_install: OPENOCD_REV     := cf1418e9a85013bbf8dbcc2d2e9985695993d9f4
-openocd_git_install: OPENOCD_OPTIONS := --enable-maintainer-mode --prefix="$(OPENOCD_DIR)" --enable-buspirate --enable-stlink
+.PHONY: openocd_install
+
+openocd_install: | $(DL_DIR) $(TOOLS_DIR)
+openocd_install: OPENOCD_URL     := git://git.code.sf.net/p/openocd/code
+openocd_install: OPENOCD_REV     := cf1418e9a85013bbf8dbcc2d2e9985695993d9f4
+openocd_install: OPENOCD_OPTIONS := --enable-maintainer-mode --prefix="$(OPENOCD_DIR)" --enable-buspirate --enable-stlink
 
 ifeq ($(OPENOCD_FTDI), yes)
-openocd_git_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --enable-ft2232_libftdi
+openocd_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --enable-ft2232_libftdi
 endif
 
 ifeq ($(UNAME), Darwin)
-openocd_git_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --disable-option-checking
+openocd_install: OPENOCD_OPTIONS := $(OPENOCD_OPTIONS) --disable-option-checking
 endif
 
-openocd_git_install: openocd_clean
-
-
+openocd_install: openocd_clean
         # download the source
 	$(V0) @echo " DOWNLOAD     $(OPENOCD_URL) @ $(OPENOCD_REV)"
 	$(V1) [ ! -d "$(OPENOCD_BUILD_DIR)" ] || $(RM) -rf "$(OPENOCD_BUILD_DIR)"

